File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        Queue<Integer> que = new ArrayDeque<>();

        for (int i = 1; i < n + 1; i++) {
            que.add(i);
        }

        while (que.size() > 1) {
            for (int i = 0; i < k - 1; i++) {
                que.add(que.poll());
            }
            que.poll();
        }

        return que.poll();
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        que = deque([i for i in range(1, n + 1)])

        while len(que) > 1:
            que.rotate(1 - k)
            que.popleft()

        return que[0]